### ISSUE TITLE
Renaming File Handling API

### ISSUE THREAD

**Issue by dfabulich** [WICG/file-handling#70]
The name "File Handling API" doesn't convey what sort of handling is being done; it sounds too similar to other APIs that allow file-based operations, including the File System API, File System Access API, etc. They all let you handle files, and FSA even includes "handle" options.

I strongly suggest renaming this API. As an alternative, I suggest calling this the "Associated File Types API."

**Comment by tomayac** [comment]
(For broader context, this started in the context of https://github.com/WICG/file-system-access/issues/342 and a [Twitter thread](https://twitter.com/dfabu/status/1471235186670981121).)

**Comment by tomayac** [comment]
Another alternative could be "File Association API". One issue I see is that the [`LaunchQueue`](https://github.com/WICG/file-handling/blob/main/explainer.md#launch) aspects are shared by [launch handling](https://github.com/WICG/sw-launch/blob/main/launch_handler.md), which likewise needs [`LaunchQueue`](https://github.com/WICG/sw-launch/blob/main/launch_handler.md#proposal).

**Comment by dfabulich** [comment]
I think it will be useful to have the word "types" in the name somewhere. The idea is that a PWA can associate itself with a _type_ of files (PNGs) rather than any individual file or directory.

I also think using a verb, "Associated," is clearer than "Association." With "Associated File Types," it's clear that the file types are being associated with something. With "File Type Association," it's not as clear… in particular it's ambigous as to whether the file types are associated with _each other_. "PNGs and JPGs are both types of images…")

**Comment by tomayac** [comment]
File association is a common term: _"In computing, a file association associates a file with an application capable of opening that file. More commonly, a file association associates a class of files (usually determined by their filename extension, such as `.txt`) with a corresponding application (such as a text editor)."_—https://en.wikipedia.org/wiki/File_association. "File *Type* Association API" seems good to me, too.

**Comment by mgiuca** [comment]
I originally named this, and I don't remember exactly why I chose "Handling" over "Association". In computing, this concept is always called either "File Handling" or "File [Type] Association". I could go with either, and don't have a strong preference for one over the other (other than sticking to what it's already called for simplicity).

I think "handler" is probably a more internally consistent term. We already have a concept of ["protocol handlers"](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler), and soon "launch handler" as a generic umbrella for all of these. We probably should have called Web Share Target "Web Share Handler" for consistency. Oh well.

It also has the benefit of being descriptive of the app itself. You have a "File Handling API"; the apps themselves are "File Handlers". If you changed it to "File Association API", what would you call the apps that implement this?

> "Associated File Types API."

I wouldn't know what that means out of context. "File Type Association" or just "File Association" is clear, if not from first principles, from 30 years of common usage.

> I also think using a verb, "Associated," is clearer than "Association." With "Associated File Types," it's clear that the file types are being associated with something. With "File Type Association," it's not as clear… in particular it's ambigous as to whether the file types are associated with each other.

That's funny, I have exactly the opposite read. "Associated File Types" suggests it relates to file types that are associated with one another. "File Type Association" is unambiguously [this thing](https://en.wikipedia.org/wiki/File_association).

I understand that "File Handling" and "File Handle" are very close and possibly confusable. But in a way, that's the point, because they are very closely related.

- A file _handle_ is the thing you get that represents a file.
- A file _handler_ is an app that can be given a file handle.

(Aside: this API actually gives you a [FileSystemHandle](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemHandle), not a [FileHandle](https://developer.mozilla.org/en-US/docs/Web/API/File_Handle_API), which is deprecated; that particular piece of confusion is orthogonal to the naming of this particular API.)

So I think those two terms being so close is less confusing and more informative. If they were totally unrelated, I would be more worried.

**Comment by evanstade** [comment]
I agree with mgiuca@. Naming is hard, but I don't think the spec should be renamed because it should match the precedent set by:

-  [protocol handlers](https://wicg.github.io/manifest-incubations/#protocol_handlers-member) (which is conceptually very similar)
-  [launch_handler](https://web.dev/launch-handler/) (which File Handling relies on heavily)

Colloquially or in documentation such as web.dev it would be fine to refer to this as enabling "file type associations" since that's an established term as Thomas points out.
